### PR TITLE
test(freefax): add standalone tests

### PR DESCRIPTION
**Pull request by cbourgois**
### Description of the Change

03c6fdd — test(freefax): add standalone tests

* add test with navigation between states in freefax modules
* add tests to check if translations are present when used (`data-translate`and `data-translate-attr` directives and `$translate.instant` call)
* Some `data-e2e-id` in html for tests purpose (see [Cypress Best Practices](https://docs.cypress.io/guides/references/best-practices.html#Selecting-Elements))
* add missing translations

### CODE REVIEW

File: packages/manager/freefax/app/cypress/integration/order.spec.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    cy.fixture('2api_freefax_details').as('fixture_2api_freefax_details');
+
+    cy.server();
+    cy.route(
+      'GET',
+      '/engine/apiv6/me',
+      'fixtures:me',
+    ).as('apiMe');
+
+    // cy.route('GET', `/engine/2api/freefax/${serviceId}/details`, 'fixtures:2api_freefax_details')
```
Comment (FredericEspiau) [inline packages/manager/freefax/app/cypress/integration/order.spec.js]: Remove this

File: packages/manager/freefax/app/cypress/integration/standalone.spec.js [hunk 2]
```diff
@@ -0,0 +1,166 @@
+import Page from '../support/po';
+
+const { DashboardPage, NotificationsPage, VoicemailConfigurationPage } = Page.FreefaxPage;
+// import translations from '../../../src/translations/Messages_fr_fR.json';
```
Comment (FredericEspiau) [inline packages/manager/freefax/app/cypress/integration/standalone.spec.js]: Delete

File: packages/manager/freefax/src/freefax.html [hunk 3]
```diff
@@ -1,7 +1,8 @@
 <section
     class="telecom-legacy telecom-freeFax-section"
-    data-ui-view>
-
+    data-ui-view
+    data-e2e-id="freefax:section">
+    <test-cyrill></test-cyrill>
```
Comment (FredericEspiau) [inline packages/manager/freefax/src/freefax.html]: What's that ?

File: packages/manager/freefax/app/cypress/support/po/base.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+export default class BasePage {
+  constructor() {
+    this.mainSection = 'body';
+    this.hashUrl = '#!/';
+  }
+
+  // eslint-disable-next-line
```
Comment (FredericEspiau) [inline packages/manager/freefax/app/cypress/support/po/base.js]: What's the disabled rule ?

File: packages/manager/freefax/app/cypress/integration/standalone.spec.js [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+// console.log(Object.keys(translations));
```
Comment (marie-j) [inline packages/manager/freefax/app/cypress/integration/standalone.spec.js]:
```suggestion

```

File: packages/manager/freefax/app/cypress/integration/standalone.spec.js [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+
+    cy.server();
+    cy.route(
+      'GET',
+      '/engine/apiv6/me',
+      'fixtures:me',
+    ).as('apiMe');
+
+    // cy.route('GET', `/engine/2api/freefax/${serviceId}/details`, 'fixtures:2api_freefax_details')
+    //   .as('2apiFreefaxDetails');
```
Comment (marie-j) [inline packages/manager/freefax/app/cypress/integration/standalone.spec.js]: Should be removed

File: packages/manager/freefax/app/cypress/integration/order.spec.js [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    dashboard.isVisible();
+
+    cy.wait([
+      '@apiMe',
+      '@2apiFreefaxDetails',
+      '@apiv6FreefaxVoicemailRouting',
+    ]);
+
+    // for test only
+    cy.get('#creditAmount').select('10');
```
Comment (marie-j) [inline packages/manager/freefax/app/cypress/integration/order.spec.js]: Why not use a `testId` like you intended to at https://github.com/ovh-ux/manager/pull/73/files#diff-02fde28b4652982e60a115a34781dd1fR157

File: packages/manager/freefax/app/cypress/support/commands.js [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+// Cypress.Commands.add("dismiss", { prevSubject: 'optional'}, (subject, options) => { ... })
+//
+//
+// -- This is will overwrite an existing command --
+// Cypress.Commands.overwrite("visit", (originalFn, url, options) => { ... })
+
+
+// Translations Commands
+// $translate.instant : call cy.createTranslateSpy() and call cy.checkTransleInstant();
+// $transalate directives : cy.checkTranslations();
```
Comment (marie-j) [inline packages/manager/freefax/app/cypress/support/commands.js]:
```suggestion
 // $translate directives : cy.checkTranslations();
```

File: packages/manager/freefax/app/cypress/support/commands.js [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+Cypress.Commands.add('createTranslateSpy', () => {
+  cy.getAngularInjector('$translate')
+    .then($translate => cy.spy($translate, 'instant').as('$translateInstantSpy'));
+});
+
+Cypress.Commands.add('checkTransleInstant', () => {
```
Comment (marie-j) [inline packages/manager/freefax/app/cypress/support/commands.js]:
```suggestion
 Cypress.Commands.add('checkTranslateInstant', () => {
```
?

File: packages/manager/freefax/app/cypress/fixtures/2api_freefax_details.json [hunk 10]
```diff
@@ -0,0 +1,37 @@
+{
+  "number": "0033123456789",
+  "fromName": "Service télécopie",
+  "redirectionEmail": [
+    "john@doe.com"
```
[the remark this answers is not in the document]
Comment (cbourgois) [reply packages/manager/freefax/app/cypress/fixtures/2api_freefax_details.json]: I use the same infos from https://github.com/ovh-ux/manager/blob/develop/packages/manager/overthebox/app/cypress/fixtures/me.json

But yes, we should update it.

File: packages/manager/freefax/app/cypress/fixtures/2api_freefax_details.json [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+      }
+    ],
+    "fullGreetingSoundId": null,
+    "temporaryGreetingSoundId": null,
+    "unreadMessages": 0,
+    "audioFormat": "mp3",
+    "annouceMessage": "",
+    "keepMessage": false,
+    "doNotRecord": false,
+    "fromName": "Service messageriee"
```
[the remark this answers is not in the document]
Comment (cbourgois) [reply packages/manager/freefax/app/cypress/fixtures/2api_freefax_details.json]: Should update it to En (relative to comments)

File: packages/manager/freefax/app/cypress/integration/standalone.spec.js [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+    dashboard.isVisible();
+    cy.createTranslateSpy();
+
+    cy.wait([
+      '@apiMe',
+      '@2apiFreefaxDetails',
+      '@apiv6FreefaxVoicemailRouting',
+    ]);
+
+    // dashboard.getMainSection()
```
[the remark this answers is not in the document]
Comment (cbourgois) [reply packages/manager/freefax/app/cypress/integration/standalone.spec.js]: Should be deleted.

File: packages/manager/freefax/app/cypress/support/po/base.js [hunk 13]
```diff
@@ -0,0 +1,32 @@
+export default class BasePage {
+  constructor() {
+    this.mainSection = 'body';
+    this.hashUrl = '#!/';
```
[the remark this answers is not in the document]
Comment (cbourgois) [reply packages/manager/freefax/app/cypress/support/po/base.js]: #62 is relative to `manager-layout-ovh`, which is not used by `manager-freefax` .

File: packages/manager/freefax/app/cypress/support/po/freefax/dashboard.js [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+import VoicemailConfigurationPage from './voicemailConfiguration';
+
+export default class DashboardPage extends BasePage {
+  constructor(serviceId) {
+    super();
+
+    this.serviceId = serviceId;
+
+    this.mainSection = 'freefax:section';
+    this.hashUrl = `#!/freefax/${this.serviceId}`;
```
[the remark this answers is not in the document]
Comment (cbourgois) [reply packages/manager/freefax/app/cypress/support/po/freefax/dashboard.js]: Tests will fail.

If we want it, we should move changes done in #62 in `manager-core`

File: packages/manager/freefax/app/cypress/support/po/freefax/notifications.js [hunk 15]
```diff
@@ -0,0 +1,18 @@
...
+import BasePage from '../base';
+
+export default class NotificationsPage extends BasePage {
+  constructor(serviceId) {
+    super();
+
+    this.serviceId = serviceId;
+
+    this.mainSection = 'freefax.notifications:section';
+    this.hashUrl = `#!/freefax/${this.serviceId}/notifications`;
```
[the remark this answers is not in the document]
Comment (cbourgois) [reply packages/manager/freefax/app/cypress/support/po/freefax/notifications.js]: Same as above.

File: packages/manager/freefax/app/cypress/support/po/freefax/voicemailConfiguration.js [hunk 16]
```diff
@@ -0,0 +1,17 @@
+
+import BasePage from '../base';
+
+export default class VoicemailConfigurationPage extends BasePage {
+  constructor(serviceId) {
+    super();
+    this.serviceId = serviceId;
+
+    this.mainSection = 'freefax.voicemail-configuration:section';
+    this.hashUrl = `#!/freefax/${this.serviceId}/voicemail`;
```
[the remark this answers is not in the document]
Comment (cbourgois) [reply packages/manager/freefax/app/cypress/support/po/freefax/voicemailConfiguration.js]: Same as above

File: packages/manager/freefax/app/cypress/fixtures/apiv6_freefax.json [hunk 17]
```diff
@@ -0,0 +1,852 @@
+{
```
[the remark this answers is not in the document]
Comment (antleblanc) [reply packages/manager/freefax/app/cypress/fixtures/apiv6_freefax.json]: In my opinion maintaining these type of files will be pretty hard.

**Comment by jleveugle** [comment]
Hello @cbourgois,
Your pull request is now closed due to inactivity.
Feel free to re-open it if you have any update.
Anna Bot for OVH UX <3

File: packages/manager/freefax/app/cypress/integration/order.spec.js (hunk 7 above)
Comment (cbourgois) [reply packages/manager/freefax/app/cypress/integration/order.spec.js]: Removed, `order.spec.js` was for test purpose.

**Comment by ovh-cds** [comment]
CDS Report build-ovh-manager_2#432.0 ✘  
* Stage 1
  * Install ✔ 
* Stage 2
  * Build ✘  
  * Tests ✔

**Comment by ovh-cds** [comment]
CDS Report build-ovh-manager_2#435.0 ✘  
* Stage 1
  * Install ✔ 
* Stage 2
  * Build ✔ 
  * Tests ✘

**Pull request merged**